answer_questions.py

```diff
@@ -28,26 +28,54 @@ def __init__(self, text_file, question_file):
         question_generator = ask.QuestionGenerator()
 
         self.wh_questions = question_generator.generateWhQuestions(processed_doc)
-        print(self.wh_questions)
+        # print(self.wh_questions)
 
-
-        q_file = open(text_file,"r+")
-        self.text = q_file.read()
-        #self.text = nlp2(self.text)
-        #self.text = self.text.decode('utf8').upper().encode('ascii')
-        doc = sentences(self.text)
-        self.tokenized_text = []
+        doc = sentences(text)
         self.sentences = [sent.string.strip() for sent in doc.sents]
 
+        with open(question_file, 'r') as q_file:
+            self.q_list = q_file.read().split('\n')
+
+    def find_best_matches_vectorized(self, question):
+
+        q_type = self.classify(question)
+
+        question = nlp(question)
+        question = ' '.join([token.text for token in question if not token.is_stop])
+        print('STOPWORDS REMOVED: ', question)
 
-        # this sets up the questions
-        q_file = open(question_file,"r+")
-        self.questions = q_file.read()
-        #self.questions = self.questions.decode('utf8').upper().encode('ascii')
-        self.q_list = self.questions.split('\n')
+        # generate (similarity, question) list for each question
+        candidates = []
+        for (sent, q_class, correct_answer) in self.wh_questions:
```

Comment:
@amyzhang17 this is the line defining the tuple

Reply:
@max you could just revise to this:
```suggestion
        for question in self.wh_questions:
            sent = question.q_string
            q_class = question.q_class
            correct_answer = question.q_answer
            ...
```

Reply:
Yeah, that looks good I'll make that change.